ql/patterns/lazyobject.hpp

```diff
@@ -114,13 +114,34 @@ namespace QuantLib {
         //@}
 
       protected:
-        mutable bool calculated_ = false, frozen_ = false, alwaysForward_ = true;
+        mutable bool calculated_ = false, frozen_ = false, alwaysForward_ = true, updating_ = false;
```

Comment:
Can't we make `updating_` private? Or is there a use case where we want to update it?